### DISCUSSION TITLE
Official macOS Silicon Build

### DISCUSSION THREAD

**Discussion by aidan-gibson** [pawelsalawa/sqlitestudio#5421] [General]
An official macOS silicon build would be nice.

kinda related https://github.com/pawelsalawa/sqlitestudio/issues/5002

**Comment by pawelsalawa** [comment]
I believe @tuffnatty did introduce it some time ago in the master branch - it's not yet released, but available in weekly builds: https://github.com/pawelsalawa/sqlitestudio/actions/runs/10987004936

**Comment by JacksonBailey** [comment]
[sqlitestudio-3.4.4-macos11.arm64.dmg](https://github.com/pawelsalawa/sqlitestudio/actions/runs/11207968646/artifacts/2021898665) worked for me. I think I needed to be signed in to download it though, just a heads up for folks landing here.

ARM. (Keyword to help this show up in searches.)

**Comment by aidan-gibson** [comment]
Sorry for getting back to this so late. Thank you for fixing this! However, it not being in Releases is preventing distribution. The [homebrew](https://formulae.brew.sh/cask/sqlitestudio) listing has an ugly unfortunate warning about needing rosetta.
I've attached their build file below. It's just ripping directly from gh releases; I believe there are rules against changing the code to rip from workflow artifacts instead. The [nixpkg](https://github.com/NixOS/nixpkgs/blob/cd1cec0295a36f7fe735fde4857ac79cbe6eac86/pkgs/by-name/sq/sqlitestudio/package.nix#L13) works similarly. 

When you get the chance, if you could update your CI/CD to push macOS ARM builds to Releases, that would be immensely helpful. I wouldn't mind fixing the brew and nix packages as well. Thank you!
```
cask "sqlitestudio" do
  version "3.4.17"
  sha256 "72dd8f8cd6f564fbdb07488d977afed5a125e08ea8d649a9756789ee21367ea4"

  url "https://github.com/pawelsalawa/sqlitestudio/releases/download/#{version}/sqlitestudio-#{version}-macos-x64.dmg",
      verified: "github.com/pawelsalawa/sqlitestudio/releases/download/"
  name "SQLiteStudio"
  desc "Create, edit, browse SQLite databases"
  homepage "https://sqlitestudio.pl/"

  app "SQLiteStudio.app"

  zap trash: [
    "~/.config/sqlitestudio",
    "~/Library/Preferences/pl.com.salsoft.SQLiteStudio.plist",
    "~/Library/Preferences/SalSoft/SQLiteStudio",
    "~/Library/Saved Application State/com.yourcompany.SQLiteStudio.savedState",
    "~/Library/Saved Application State/pl.com.salsoft.SQLiteStudio.savedState",
  ]

  caveats do
    requires_rosetta
  end
end
```

**Comment by pawelsalawa** [comment]
MacOS ARM will be officially supported with version 4.0.0 onwards. It will actually be an universal binary (both x86_64 and ARM).

**Reply by cbertelli** [reply]
I understant. I had a working version a week ago, but it was overwritten by a `brew upgrade` and I cannot find it anymore on web. Can you help me find it again (I think it has been deleted) or suggest a recent build? Sorry for bothering you

**Comment by jd-terry** [comment]
I have downloaded the mac version (3.4.20). I am simply going through a tutorial from an older sql book and cannot load the "weather_stations" database. Getting errors on libsqlite3.dylib regarding logging persist as well as /private/var/db/DetachedSignatures. Would be happy to attempt using beta version if helpful. Running Tahoe 26.2 on an M4.

**Reply by pawelsalawa** [reply]
Where do you keep your weather_stations database file? Is it a writable folder?

**Comment by jd-terry** [comment]
I downloaded a zip file, unzipped it and it is in the directory, owned by me with read/write permission.

**Reply by pawelsalawa** [reply]
I'm not sure what does it mean that you are:

> Getting errors on libsqlite3.dylib regarding logging persist as well as /private/var/db/DetachedSignatures.

Can you show me some screenshot of where these errors appear? Is it inside of SQLiteStudio window, or from the macOS?

**Comment by jd-terry** [comment]
They are out of my system log on the mac.  Here is an example:

2026-01-10 16:55:49.311499-0700 0x9979b    Error       0x21a1aa             414    0    tccd: (libsqlite3.dylib) [com.apple.libsqlite3:logging-persist] cannot open file at line 51043 of [f0ca7bba1c]
2026-01-10 16:55:49.311500-0700 0x9979b    Error       0x21a1aa             414    0    tccd: (libsqlite3.dylib) [com.apple.libsqlite3:logging-persist] os_unix.c:51043: (2) open(/private/var/db/DetachedSignatures) - No such file or directory

I was out for a bit -- back now if you need anything else.

**Comment by jd-terry** [comment]
pawaelsalawa --

I had to re-install sqlitestudio to get the errors in the log to show you.  I also had to reload the data. I attempted to load the database and it now seems to be working -- I will check the logs to see there are any errors being thrown.

Thanks you for your attention, I will let you know if I have any further issues.

:)
